parsl/providers/lsf/lsf.py

```diff
@@ -120,15 +121,15 @@ def _status(self):
             parts = line.split()
             if parts and parts[0] != 'JOBID':
                 job_id = parts[0]
-                status = translate_table.get(parts[2], 'UNKNOWN')
-                self.resources[job_id]['status'] = status
+                state = translate_table.get(parts[2], JobState.UNKNOWN)
+                self.resources[job_id]['status'] = JobStatus(state)
                 jobs_missing.remove(job_id)
 
         # squeue does not report on jobs that are not running. So we are filling in the
         # blanks for missing jobs, we might lose some information about why the jobs failed.
         for missing_job in jobs_missing:
-            if self.resources[missing_job]['status'] in ['PENDING', 'RUNNING']:
-                self.resources[missing_job]['status'] = 'COMPLETED'
+            # if self.resources[missing_job]['status'] in ['PENDING', 'RUNNING']:
```

Comment:
That opens a can of worms, since you can never guarantee that a FAILED job will not eventually complete. So you're setting it up for hanging.

You might want to consider having a separate job, which is effectively what happens if "job" means an attempt to run something, or having a FAILED_BUT_WILL_RETRY state.